packages/vm/src/environment.rs

```diff
@@ -301,13 +301,18 @@ impl<A: BackendApi, S: Storage, Q: Querier> Environment<A, S, Q> {
 
     pub fn memory(&self) -> Memory {
         self.with_wasmer_instance(|instance| {
-            let mut memories: Vec<Memory> = instance
+            let first: Option<Memory> = instance
                 .exports
                 .iter()
```

Comment:
On second thought, I guess it's for generality / API consistency, but why have an iterator for something that can contain at most one element? I mean, maybe this can be simplified further (at the risk of losing generality).

Reply:
The `instance` is a `wasmer::Instance` we don't control. In the Wasmer API memories are modelled as a list because future iterations of the Wasm spec probably allow you to have more than one memory ([see multi-memory proposal](https://github.com/WebAssembly/multi-memory/blob/master/proposals/multi-memory/Overview.md)). The binary Wasm file format is already prepared for that. The first version of Wasm (MVP) only supports one memory and this is enforced somewhere internally in Wasmer.